### PR TITLE
test(discover-v2): Add tests for QueryBuilder

### PR DESCRIPTION
**Pull request by leedongwei**
This depends on #14791 to be merged into `master`.

### CODE REVIEW

**Review by markstory — APPROVED** [review]

File: tests/js/spec/views/eventsV2/utils.spec.jsx [hunk 1]
```diff
@@ -240,3 +244,274 @@ describe('getFieldRenderer', function() {
...
+      star: {
+        trek: 'maybe',
+        wars: 'perhaps',
+      },
+    },
+  };
+  let browserHistoryPush;
+
+  beforeAll(() => {
+    browserHistoryPush = browserHistory.push;
```
Comment (billyvg) [inline tests/js/spec/views/eventsV2/utils.spec.jsx]: No need to do this, you can call `mockRestore` on the jest mock function to restore the original implementation

**Review by billyvg — APPROVED** [review]

File: tests/js/spec/views/eventsV2/utils.spec.jsx (hunk 1 above)
Comment (markstory) [reply tests/js/spec/views/eventsV2/utils.spec.jsx]: Doesn't our jest config also automatically mock out this function too?

**Comment by leedongwei** [comment]
Latest force-push has no code changes. 

The branch was outdated, so it was rebased against `master` and conflicts were resolved.

**Comment by leedongwei** [comment]
Due to changes in 00622ec, we need to rename `alias` to `fieldnames` in 3456ad3. There's no code logic changes so I will not request a new review. 

😬

**Pull request merged**